WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderForm.swift

```diff
@@ -148,6 +148,11 @@ struct OrderFormPresentationWrapper: View {
                     .accessibilityIdentifier(OrderForm.Accessibility.cancelButtonIdentifier)
                 },
                 isShowingSecondaryView: $viewModel.isProductSelectorPresented)
+            .onChange(of: horizontalSizeClass) { _ in
+                // Save: in-flight order details when there's size class changes
+                viewModel.saveInflightCustomerDetails()
+                viewModel.saveInFlightOrderNotes()
+            }
```

Comment:
I moved this here, to fix the UI test – this avoids us calling these methods when the view first appears.

Reply:
This update wasn't working on my end, I'm not entirely sure why, but the `.onChange` was never called upon size class changes so the methods weren't called either. 

I suspect it could be due to the ViewBuilder redrawing the whole thing, which wouldn't trigger the .onChange? After some experimentation it does work properly if we add it here instead: https://github.com/woocommerce/woocommerce-ios/pull/12306/commits/8a9bbc97bf29e8963775db3cd9f345890b22c094

Reply:
Hmm.

That seems plausible, except it must have been working for me, or I wouldn't have put the code there. Very strange.

Where you've moved it, you don't need to use the `horizontalSizeClass`, as you have `presentationStyle` already.

Reply:
> Where you've moved it, you don't need to use the horizontalSizeClass, as you have presentationStyle already.

That was my original approach, but it also happens here that if we switch the `.onChange(of: horizontalSizeClass)` for `.onChange(of: presentationStyle)`, the `onChange()`  is never called when there's an update, and the in-flight data not saved. 

I've wonder if could be related to the usage of `@Environment` for the custom presentationStyle value, and if the behaviour is slightly different than other wrappers when updating/publishing values. I found comments regarding different behaviour and the need to use view modifiers like `.environment(\.key, value)` or changes might not propagate as expected, but at this point everything got a bit hairy to debug.